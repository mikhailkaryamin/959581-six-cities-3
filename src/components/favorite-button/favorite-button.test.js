import React from 'react';
import renderer from 'react-test-renderer';
import {
  AuthorizationStatus
} from '../../reducer/user/user';
import {
  FavoriteButton
} from './favorite-button';

const onFavoriteClick = () => {};
const authStatus = AuthorizationStatus.AUTH;

test(`Render correctly favorite button`, () => {
  const tree = renderer.create(
      <FavoriteButton
        id={1}
        classModificator={`place-card`}
        width={25}
        height={25}
        isFavorite={true}
        onFavoriteClick={onFavoriteClick}
        authStatus={authStatus}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
