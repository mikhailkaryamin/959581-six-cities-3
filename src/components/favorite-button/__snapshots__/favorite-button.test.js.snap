// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly favorite button 1`] = `
<button
  className="place-card__bookmark-button button place-card__bookmark-button--active"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height={25}
    width={25}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
